### PR TITLE
changes to elo algorithm for precision

### PR DESCRIPTION
**Pull request by Capn05**
Most changes are in response to comments on previous two pull requests. 
To address the issues Kevin and I posed in Pull Request 16 (https://github.com/Gabriels-Support-Team/Capstone/pull/16):

I have implemented a strategy where only the movies that are already in the system have their ratings adjusted using the Elo formula. This adjustment occurs based on the outcomes of user comparisons, where movies can either gain or lose rating points depending on whether they win or lose these comparisons.
For new movies being introduced into the system, we assign ratings differently. Instead of using the Elo formula, the rating for a new movie is set to be the average of the ratings of the two movies that currently define the bounds of its comparison range. This method ensures that the rating of a new movie accurately reflects the position it should occupy within the existing rankings, as determined by user comparisons.
Furthermore, to enhance the accuracy and responsiveness of the rating system, I plan to dynamically adjust the K-factor based on the number of comparisons a movie undergoes. This dynamic adjustment is designed to help movie ratings "converge" to their true value over time. By allowing the K-factor to change with the frequency of comparisons, I aim to address and mitigate issues of rating clustering. This dynamic approach to adjusting the K-factor is anticipated to be effective enough to replace the scaling function initially intended to handle clustering. 

Other changes are addressing syntactical/style issues brought up in pull request 15 and 16

Video Below:
In the video I forgot to mention that as the comparisons are being made, tarzan and star wars lose and gain rating respectively, showcasing the Elo change
https://www.loom.com/share/d32a17405f5242e8bd254545f03a9695

### CODE REVIEW

File: src/RankMovie.jsx [hunk 1]
```diff
@@ -69,39 +68,105 @@ function RankMovie() {
...
     winnerNewRating = Math.max(1, Math.min(10, winnerNewRating));
     loserNewRating = Math.max(1, Math.min(10, loserNewRating));
 
-    try {
+    setLow(newLow);
+    setHigh(newHigh);
+
+    // Update ratings for both winner and loser
+    if (prefersNewMovie){
+    await updateMovieRatings(winner.movieId, newDynamicRating);
```
Comment (BriptimusPrimus) [inline src/RankMovie.jsx:152]: This line is not indented. Not sure if it's the only one, but these type of issues should be resolved with the linter. Maybe we can even hook it up to always run when a commit is being made.

File: src/App.jsx [hunk 2]
```diff
@@ -34,55 +34,55 @@ function App() {
               path="/recommendations"
               element={
                 <ProtectedRoute>
-                  <Recommendations></Recommendations>
+                  <Recommendations />
```
Comment (BriptimusPrimus) [inline src/App.jsx:37]: 👍

File: scripts/loadMovies.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ const prisma = new PrismaClient();
 function loadMovies() {
   const data = fs.readFileSync("../ml-1m/movies.dat", "utf-8");
   const movies = data.split("\n");
-  let promises = movies.map((movie) => {
+  const promises = movies.map((movie) => {
```
Comment (BriptimusPrimus) [inline scripts/loadMovies.js:9]: :+1

**Comment by BriptimusPrimus** [comment]
Is there a reason to keep the calculation on the front end?

File: src/firebase/auth.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ export const doSignInWithEmailAndPassword = async (email, password) => {
     );
     return userCredential.user;
   } catch (error) {
-    throw error;
+    console.error("Error signing in with email and password:", error);
```
Comment (BriptimusPrimus) [inline src/firebase/auth.js:20]: We are expected not to have console logs in the committed code. It's fine to use them for debugging during dev, but it should not make it to the PR.
If what you need is the user to have visibility of the errors, then they should be displayed in the UI.

File: src/RankMovie.jsx [hunk 5]
```diff
@@ -69,39 +68,105 @@ function RankMovie() {
   }
 
   const handleComparison = async (prefersNewMovie) => {
-    const winner = prefersNewMovie ? newUserMovie : userMovies[mid];
-    const loser = prefersNewMovie ? userMovies[mid] : newUserMovie;
+    //get newly logged movie, initial
+    if (!currentUser || !newMovie || mid === undefined) {
+      console.error("Current user or new movie data is not available.");
```
Comment (samyalihamad) [inline src/RankMovie.jsx:73]: Make sure not to use `console.log` for messages to the user. You can use it during development just remove them before checking in the code.

**Pull request merged**